src/components/CustomSidebar.jsx

```diff
@@ -0,0 +1,123 @@
+           <Typography variant="overline" >Partner code: IA234522</Typography>
+           <Typography variant="overline" >Referral code: IA234522</Typography>
+        </Box> 
+        <Box display="flex" justifyContent="right">
+          <Divider width='242px'  />
+        </Box>
+            <MenuItem className="list-gap" active>   
+            <img src={Dashboard} className="image" alt="Dashboard Icon" /> 
+              <Typography sx={styles.hover} variant="body2">Dashboard</Typography>
+            </MenuItem>
+             <MenuItem className="list-gap" active>   
+            <img src={Leads} className="image" alt="Dashboard Icon" /> 
+              <Typography sx={styles.hover} variant="body2">Leads</Typography>
+            </MenuItem>
+             <MenuItem className="list-gap" active>   
+            <img src={Buyers} className="image" alt="Dashboard Icon" /> 
+              <Typography sx={styles.hover} variant="body2">Buyers</Typography>
+            </MenuItem>
+             <MenuItem className="list-gap" active>   
+            <img src={Clients} className="image" alt="Dashboard Icon" /> 
+              <Typography sx={styles.hover}  variant="body2">Clients</Typography>
+            </MenuItem>
+             <MenuItem className="list-gap" active>   
+            <img src={Properties} className="image" alt="Dashboard Icon" /> 
+              <Typography sx={styles.hover} variant="body2">Properties</Typography>
+            </MenuItem>
+             <MenuItem className="list-gap" active>   
+            <img src={Orders} className="image" alt="Dashboard Icon" /> 
+              <Typography sx={styles.hover} variant="body2">Orders</Typography>
+            </MenuItem>
+             <MenuItem className="list-gap" active>   
+            <img src={RedBox} className="image" alt="Dashboard Icon" /> 
+              <Typography sx={styles.hover} variant="body2">Red Box</Typography>
+            </MenuItem>
+             <MenuItem className="list-gap" active>   
+            <img src={CallBack} className="image" alt="Dashboard Icon" /> 
+              <Typography sx={styles.hover} variant="body2">Call Back</Typography>
+            </MenuItem>
+            <MenuItem className="list-gap" active>   
+            <img src={Investments} className="image" alt="Dashboard Icon" /> 
+              <Typography sx={styles.hover} variant="body2">Investments</Typography>
+            </MenuItem>
+             <MenuItem className="list-gap" active>   
+            <img src={Payment} className="image" alt="Dashboard Icon" /> 
+              <Typography sx={styles.hover} variant="body2">Payments</Typography>
+            </MenuItem>
+             <MenuItem className="list-gap" active>   
+            <img src={Magik} className="image" alt="Dashboard Icon" /> 
+              <Typography sx={styles.hover} variant="body2">Magik</Typography>
+            </MenuItem>
+          </Menu>
```

Comment:
This code from line number 41 to 84, you have repeated the menu items multiple time, the code is almost same, only image and menu item is changing, so create an array like this

```
const menuItems = [
  { src: Dashboard, label: "Dashboard"},
  { src: Leads, label: "Leads" },
  { src: Buyers, label: "Buyers" },
  { src: Clients, label: "Clients" },
  { src: Properties, label: "Properties" },
  { src: Orders, label: "Orders" },
  { src: RedBox, label: "Red Box" },
  { src: CallBack, label: "Call Back" },
  { src: Investments, label: "Investments" },
  { src: Payment, label: "Payments" },
  { src: Magik, label: "Magik" }
];
```

and the iterate over it like this

```
{
menuItems.map((item, index) => (
  <MenuItem key={index} className="list-gap" active>   
    <img src={item.src} className="image" alt={`${item.label} Icon`} /> 
    <Typography sx={styles.hover} variant="body2">{item.label}</Typography>
  </MenuItem>
));
}
```

Reply:
Yes..thanks this looks more convenient. Will implement it